components/blueprints/BlueprintsLists.vue

```diff
@@ -0,0 +1,158 @@
+<template>
+<div class="mt-5">
+    <p class="top-breadcrumb" data-aos="fade-right">
+        Solutions
+    </p>
+    <h1 data-aos="fade-left">Blueprints</h1>
+    <h5 data-aos="fade-right">The first step is always the hardest. Explore blueprints to kick-start your next flow.</h5>
+    <div class="grid gap-3 mt-5" data-aos="fade-left">
+        <button
+            v-for="tag in tags"
+            :key="tag.name"
+            :class="{ 'active': tag.name === activeTag.name }"
+            @click="setTagBlueprints(tag)"
+            class="m-1 rounded-button"
+        >
+            {{ tag.name }}
+        </button>
+    </div>
+    <div class="row my-5">
+        <div class="row mb-4 justify-content-center">
+            <div class="col-12 col-md-6 col-lg-4">
+                <input type="text" class="form-control form-control-lg" id="search-input" placeholder="Search blueprints" v-model="searchQuery">
+            </div>
+        </div>
+        <div class="col-lg-4 col-md-6 mb-4" v-for="blueprint in blueprints" :key="blueprint.id">
+            <BlueprintsBlueprintCard :blueprint="blueprint" :icons="icons" data-aos="zoom-in" :tags="tags" />
+        </div>
+        <div class="d-flex justify-content-between">
+            <div class="items-per-page">
+                <select class="form-select" aria-label="Default select example" v-model="itemsPerPage">
+                    <option :value="10">10</option>
+                    <option :value="25">25</option>
+                    <option :value="50">50</option>
+                </select>
+            </div>
+            <div class="d-flex align-items-baseline" v-if="totalBlueprints > itemsPerPage">
+                <BlueprintsPagination :total-pages="totalPages" @on-page-change="changePage" />
+                <span class="total-pages">Total {{ totalBlueprints }}</span>
+            </div>
+        </div>
+    </div>
+</div>
+</template>
+
+<script setup>
+const currentPage = ref(1)
+const itemsPerPage = ref(25)
+const blueprints = ref([])
+const activeTag = ref({ name: 'All tags' })
+const tags = ref([])
+const props = defineProps(["icons","tags"])
+const totalPages = ref(0)
+const totalBlueprints = ref(0)
+const searchQuery = ref('')
+const route = useRoute()
+const router = useRouter()
+
+if(props.tags) {
+    tags.value = [{ name: 'All tags' }, ...props.tags]
+}
+
+const setTagBlueprints = (tagVal) => {
+    activeTag.value = tagVal
+}
+
+if(route.query.page) currentPage.value = parseInt(route.query.page)
+if(route.query.size) itemsPerPage.value = parseInt(route.query.size)
+if(route.query.tags) activeTag.value = tags.value.find(f => f.id == route.query.tags)
+if(route.query.q) searchQuery.value = route.query.q
```

Comment:
You could destructure page, size and tags first then have the conditions and the parsers to take a single variable rather than a reference